### PR TITLE
JBPM-8212: Make TimerManager independent from Drools interfaces

### PR DESCRIPTION
**Pull request by evacchi**
Initial work, let's see if we're on the right path

### CODE REVIEW

**Comment by evacchi** [comment]
This is a huge PR indeed. I have moved both TimerService and TimerManager (for the most part) to a new project `api/kie-services` with (for now)

```
org.kie.services.signal
org.kie.services.time
org.kie.services.time.impl
org.kie.services.time.manager
```


 this is mostly a refactor i.e. no redesign took place, I just limited to remove dependency cycles; e.g. TimerManager depended on ProcessInstance only for `sendSignal()` so I added a `Signalable` interface only for that method, etc.

I know it's pretty big, but please do take some time to have a look.

**Comment by evacchi** [comment]
ping @mswiderski @mariofusco please take a look when you have time
proposal: I have the same patch ready for master, we can talk about it (I added a module `kie-core` instead of `kie-services`)

File: drools/drools-core/src/main/java/org/drools/core/time/SelfRemovalJobContext.java [hunk 1]
```diff
@@ -47,9 +49,8 @@ public JobHandle getJobHandle() {
         return jobContext.getJobHandle();
     }
 
-    @Override
     public InternalWorkingMemory getWorkingMemory() {
-        return jobContext.getWorkingMemory();
+        throw new UnsupportedOperationException("delegate.getWorkingMemory()");
```
Comment (mswiderski) [inline drools/drools-core/src/main/java/org/drools/core/time/SelfRemovalJobContext.java]: is this still being used?

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightProcessRuntime.java [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+        this.processInstanceManager.clearProcessInstancesState();
+    }
+
+    public boolean isActive() {
+//        Boolean active = (Boolean) kruntime.getEnvironment().get("Active");
+//        if (active == null) {
+//            return true;
+//        }
+//
+//        return active.booleanValue();
```
Comment (mswiderski) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightProcessRuntime.java]: maybe just remove the commented code...

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightProcessRuntimeContext.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        RuleFlowProcessInstance processInstance = new RuleFlowProcessInstance();
+        processInstance.setProcess(process);
+
+        if (correlationKey != null) {
+            processInstance.getMetaData().put("CorrelationKey", correlationKey);
+        }
+
+//        ((InternalProcessRuntime) kruntime.getProcessRuntime()).getProcessInstanceManager()
+//                .addProcessInstance( processInstance, correlationKey );
```
Comment (mswiderski) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightProcessRuntimeContext.java]: work in progress? or why is this commented out?

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightProcessRuntimeServiceProvider.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    /**
+     * Adapts a ProcessInstance to the generic Signalable interface
+     */
+    private static class SignalableAdaptor implements Signalable {
+
+        private final ProcessInstance instance;
+
+        SignalableAdaptor(org.kie.api.runtime.process.ProcessInstance instance) {
```
Comment (mswiderski) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightProcessRuntimeServiceProvider.java]: why not making ProcessInstance Signalable by default?

**Review by mswiderski — COMMENTED** [review]
few classes do not have license headers so the build might be affected assuming enforcer is active

overall looks good, just small comments inline.

File: drools/drools-core/src/main/java/org/drools/core/time/SelfRemovalJobContext.java (hunk 1 above)
Comment (evacchi) [reply drools/drools-core/src/main/java/org/drools/core/time/SelfRemovalJobContext.java]: right that's pretty ugly. I wanted to decouple JobContext from their WorkingMemory dependency and that's a dangling leftover. I'll look into it

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightProcessRuntimeServiceProvider.java (hunk 4 above)
Comment (evacchi) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightProcessRuntimeServiceProvider.java]: true. Process is already an EventListener, which is the same as a Signalable. Let's drop all of this

**Pull request merged**